Add blog post about API documentation sprint (#290)
openSUSE/obs-landing: _posts/development/2021-04-12-api-documentation-remake.md

---
layout: post
title: API Documentation Remake
category: development
---

> Following the last deliverable about [API Documentation](/2021/01/26/api-documentation-remake), we focused this time on documenting some OBS API endpoints with [Swagger UI](https://swagger.io/tools/swagger-ui/).

Reviewer: Neither sprint nor deliverable makes much sense for our readers. Those are things that only concern us. Just tell them what's new :smile_cat:

> Following the last sprint about [API Documentation](/2021/01/26/api-documentation-remake), we focused our last deliverable in documenting some OBS API endpoints with [Swagger UI](https://swagger.io/tools/swagger-ui/).

Reviewer 2: ```suggestion
Following the last deliverable about [API Documentation](/2021/01/26/api-documentation-remake), we focused this time on documenting some OBS API endpoints with [Swagger UI](https://swagger.io/tools/swagger-ui/).
```

> Following the last sprint about [API Documentation](/2021/01/26/api-documentation-remake), we focused our last deliverable in documenting some OBS API endpoints with [Swagger UI](https://swagger.io/tools/swagger-ui/).

Reviewer: Neither sprint nor deliverable makes much sense for our readers. Those are things that only concern us. Just tell them what's new :smile_cat:

> In the last days we documented, using [OpenAPI specification](https://www.openapis.org/), a list of endpoints, which will grow in the next iterations. So here you have:

Reviewer 2: I would drop the link to the OpenAPI specification and just write about the endpoint groups which are now documented.

---
layout: post
title: API Documentation Remake
category: development
---

Following the last sprint about [API Documentation](/2021/01/26/api-documentation-remake), we focused our last deliverable in documenting some OBS API endpoints with Swagger UI.

In the last days we documented, using OpenAPI, a list of endpoints, which will grow in the next iterations. So here you have:

<figure>
  <img src="/images/posts/2021-04-12/endpoints.png" alt="OBS API New Documentation Page" />
  <figcaption>OBS API New Documentation Page</figcaption>
</figure>

> This documentation page is now available [here](https://build.opensuse.org/apidocs-new/).

Reviewer 2: Everything after this line is pretty much the same as the previous blog post about the API new documentation. I'm not sure if it's really needed to keep this for every blog post related to API documentation. We could have this in a partial if we decide to keep this. The partials are in https://github.com/openSUSE/obs-landing/tree/master/_includes/partials

Author: i would prefer to drop the two paragraphs:

```
OpenAPI specification is “a specification for machine-readable interface files for describing, producing, consuming, and visualizing RESTful web services” (according to Wikipedia). Once a definition based on the OpenAPI specification is created, a complete ecosystem of tools can be used. One of these tools is Swagger UI, which provides a web interface for the documentation included in the OpenAPI definition.

Swagger UI not only provides an ordered list of the API endpoints, grouped, color-coded by action, examples of parameters and responses, … It also allows you to test your API directly from your browser: asks for authentication when required, asks for the parameters needed, and shows the results of the performed request.
```

Reviewer 2: I think it's also better this way.
